### DISCUSSION TITLE
Stuck updating Crazyflie 2.1 from iOS Client

### DISCUSSION THREAD

**Discussion by jeffort** [bitcraze/discussions#738] [Q&A]
Greetings!  I am stuck trying to update my Crazyflie 2.1 from the iOS client.  It progresses loading  2 of 2 files, but then loses communication at the very end.  Now, when I boot up the device, it seems bricked with a solid blue light on M2,  I can get back to the boo loader mode by holding the power button.  I do not have the radio, is this the only way to connect from the Linux client?  I have tried using the troubleshooting guide, but nothing has helped.  Any help appreciated!

**Comment by puliceworx** [comment]
I was literally coming here to post about this.  I got my Crazyflie 2.x on Friday. Flew it a very tiny bit after going through the getting started. Then I followed getting started using the flow deck and did the update via the iOS app.  I have screenshots of the process. I can still get into the bootloader and do this process over and over and it is the same result. 
![dl_stm32_progress-90](https://github.com/bitcraze/discussions/assets/124603383/d52a4af1-27d5-4880-831f-b526ca17cd74)

First message after completion 
![first-msg-after-completion](https://github.com/bitcraze/discussions/assets/124603383/19c3d787-9823-4d59-bbab-47eff1c06c61)


Success message after clearing the first message. 
![success-msg-comes-after-first](https://github.com/bitcraze/discussions/assets/124603383/e87866be-3166-4546-86c4-3084599ebd60)

I tried to update via the client from the VM but the device does not show up via USB.  

Basically consider this bricked at the moment and cannot continue.

**Reply by knmcguire** [reply]
You can not update the crazyflie through the cfclient by connecting it with usb because of the reason we mention below. Cfclient will only work with a Crazyradio.

**Comment by knmcguire** [comment]
So, if you have a crazyradio and the client and usb devices correctly setup, then there are these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/. 

There is a way to update/recover the firmware through usb but a **big fat warning**, this can potentially remove the bootloader so be carefull, but for recovery purposes and if you don't have a crazyradio then there is no other way: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/

Just as a mention, if you want to do any type of development on the crazyflie, we do consider having the crazyradio a must.

**Reply by puliceworx** [reply]
My crazyradio and debug adapter are on the way.  

Thank you for the reply!

**Reply by puliceworx** [reply]
My radio and adapter arrived ahead of schedule :) 

Got it all fixed up.  Thank you

**Comment by jeffort** [comment]
Is there any solution for the scenario posted above without the radio and with the iOS client only?  The described flow and screenshots are identical for me.

**Reply by puliceworx** [reply]
Hi Jeff, I am assuming there is an issue with BLE packet loss.  

I just used the crazyradio to do the update like suggested and noticed the difference between the two downloads. 

When I did it with the iOS app I saw the M1 led blinking fast green with several red flashes which I assume means packet loss. 

When performing the update with crazyradio and the client the light never flashed red and always remained green. 
The other thing I noticed was speed.  As you likely saw, like I did, the iOS update took like 19 minutes.  This method it was done in under a minute. 

*Sidenote I saw you may have inquired about the debug adapter.  I'm a firmware engineer so I already had an STLInk V2 but the crazyflie is too small to host a typical jtag connector so they use a special breakout board to connect it. 

I got the radio and adapter in 5 days from Sweden by paying for the international priority mail, just FYI.

**Reply by jeffort** [reply]
Thanks, I'll give the radio a try!
